pkg/sql/catalog/table_elements.go

```diff
@@ -60,9 +60,32 @@ type TableElementMaybeMutation interface {
 	Dropped() bool
 }
 
+// ConstraintProvider is an interface for something which might unwrap
+// a constraint.
+type ConstraintProvider interface {
+
+	// AsCheck returns the corresponding CheckConstraint if the
+	// mutation is on a check constraint, nil otherwise.
```

Comment:
nit: Is the comment stale? Why do we say "...if the *mutation* is on a check constraint..."?